src/goPackages.ts

```diff
@@ -97,11 +97,21 @@ export function getAllPackages(): Promise<Map<string, string>> {
 
 /**
  * Returns mapping of import path and package name for packages that can be imported
+ * Possible to return empty if useCache options is used.
  * @param filePath. Used to determine the right relative path for vendor pkgs
+ * @param useCache. Force to use cache
  * @returns Map<string, string> mapping between package import path and package name
  */
-export function getImportablePackages(filePath: string): Promise<Map<string, string>> {
-	return Promise.all([isVendorSupported(), getAllPackages()]).then(([vendorSupported, pkgs]) => {
+export function getImportablePackages(filePath: string, useCache: boolean = false): Promise<Map<string, string>> {
+	let getAllPackagesPromise: Promise<Map<string, string>>;
+	if (useCache) {
+		// forced to use cache
+		getAllPackagesPromise = Promise.race([getAllPackages(), allPkgsCache]);
```

Comment:
When `allPkgsCache` is not set yet, then the Promise here waits for `getAllPackages()` to finish?

Reply:
It use `Promise.race`
`getAllPackages` has cache which is `allPkgsCache` and has it's own caching logic.
when `getAllPackages` use it's cache then it will selected by `Promise.race`
when not using cache then it will fallback to `allPkgsCache` which can be empty or outdated cache that is why I'm using `Promise.race([..., allPkgsCache])`

Reply:
it always return 'fast'
```js
function timeout(millis) {
  return new Promise((resolve, reject) => {
    setTimeout(resolve, millis);
  });
}

function fastResponse() {
  return new Promise((resolve, reject) => {
    resolve('fast');
  });
}

function slowResponse() {
  return timeout(500).then(() => 'slow');
}

const p = Promise.race([slowResponse(), fastResponse(), 'instant']);
p.then((v) => console.log(v));
```